SA3/qa_web/views.py

```diff
@@ -172,19 +172,41 @@ def answers(request, id_):
          initialSelectValue = "highestScore"
          q_answers = Answers.objects.filter(question=q, correct_answer=False).annotate(points=F('upvotes')-F('downvotes')).order_by('-points')
 
-
     q_best_answer = Answers.objects.filter(question=q, correct_answer=True)
     q_comments = Comments.objects.filter(question=q)
     a_comments = Comments.objects.filter(answer__question=q)
 
+    # Voting indicators
+    if request.user.is_authenticated:
+        vote_on_q = Vote.objects.filter(user=request.user, question=q)
+        votes_on_answers = Vote.objects.filter(user=request.user, answer__in=q_answers)
+        vote_on_best_answer = Vote.objects.filter(user=request.user, answer_id=q_best_answer[0].id if len(q_best_answer) else -1)
+        votes_on_q_comments = Vote.objects.filter(user=request.user, comment__in=q_comments)
+        votes_on_a_comments = Vote.objects.filter(user=request.user, comment__in=a_comments)
+        pos_vote_q = len(vote_on_q) and vote_on_q[0].positive
+        neg_vote_q = len(vote_on_q) and not vote_on_q[0].positive
+
+        pos_vote_a = [v.answer.id for v in list(votes_on_answers) + list(vote_on_best_answer) if v.positive]
```

Comment:
Nice way to gather all the votes for all answers in the question. Same can be said for all comments in the question